src/components/checkbox/checkbox-group/checkbox-group.test.tsx

```diff
@@ -0,0 +1,132 @@
+  expect(screen.getByText("label-2")).toBeVisible();
+});
+
+test("should render with the provided legend", () => {
+  render(
+    <CheckboxGroup legend="legend">
+      <Checkbox value="1" label="label" onChange={() => {}} />
+    </CheckboxGroup>
+  );
+
+  expect(screen.getByText("legend")).toBeVisible();
+});
+
+test("should render legendHelp as a hint text when validationRedesignOptIn is true", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <CheckboxGroup legend="legend" legendHelp="legendHelp">
+        <Checkbox value="1" label="label" onChange={() => {}} />
+      </CheckboxGroup>
+    </CarbonProvider>
+  );
+
+  expect(screen.getByText("legendHelp")).toBeVisible();
+});
+
+test("should render required checkbox children when required prop is set", () => {
+  render(
+    <CheckboxGroup legend="legend" required>
+      <Checkbox value="1" label="label-1" onChange={() => {}} />
+      <Checkbox value="2" label="label-2" onChange={() => {}} />
+    </CheckboxGroup>
+  );
+
+  const checkboxes = screen.getAllByRole("checkbox");
+  expect(checkboxes[0]).toBeRequired();
+  expect(checkboxes[1]).toBeRequired();
+});
+
+it("should append (optional) text on the legend when isOptional prop is set", () => {
+  render(
+    <CheckboxGroup legend="legend" isOptional>
+      <Checkbox value="1" label="label-1" onChange={() => {}} />
+      <Checkbox value="2" label="label-2" onChange={() => {}} />
+    </CheckboxGroup>
+  );
+
+  expect(screen.getByText("legend")).toHaveStyleRule(
+    "content",
+    '"(optional)"',
+    { modifier: "::after" }
+  );
+});
+
+test("should render error message and no tooltip when error prop is set and validationRedesignOptIn is true", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <CheckboxGroup legend="legend" error="error">
+        <Checkbox value="1" label="label" onChange={() => {}} />
+      </CheckboxGroup>
+    </CarbonProvider>
+  );
+
+  expect(screen.getByText("error")).toBeVisible();
+  expect(screen.queryByRole("button")).not.toBeInTheDocument();
+});
+
+test("should render warning message and no tooltip when warning prop is set and validationRedesignOptIn is true", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <CheckboxGroup legend="legend" warning="warning">
+        <Checkbox value="1" label="label" onChange={() => {}} />
+      </CheckboxGroup>
+    </CarbonProvider>
+  );
+
+  expect(screen.getByText("warning")).toBeVisible();
+  expect(screen.queryByRole("button")).not.toBeInTheDocument();
+});
+
+test("should render with expected styles when legendInline is true", () => {
+  render(
+    <CheckboxGroup legend="legend" legendInline>
+      <Checkbox
+        data-role="checkbox-1"
+        value="1"
+        label="label-1"
+        onChange={() => {}}
+      />
+      <Checkbox value="2" label="label-2" onChange={() => {}} />
+    </CheckboxGroup>
+  );
+
+  // get styles set by :first-child
```

Comment:
question: can we not use `toHaveStyle` here as it asserts against the computed styles?

Reply:
You are absolutely right and I have no idea why I made it harder for myself by doing this